File: python/prime_factors.py
# The challenge is to break down a given number into its prime
#  factors. For example, if the input number is 24, we need to find all 
# the prime factors that multiply together to equal 24. In this case,
#  24 can be factored as 2 × 2 × 2 × 3, because 2 and 3 are prime numbers, 
# and their product gives 24.


def prime_factors(n):
    factors = [] 
    while n>1:
        if n%2==0:
            factors.append(2)
            n=n//2
        elif n%3==0:
            factors.append(3)
            n=n//3
        else:
            factors.append(n)
            break
    return factors
    
factors = prime_factors(15)
print(factors)

File: golang/prime_factors/prime_factors.go
package main

import "fmt"

func prime_factors(num int) []int {

	factors := []int{}

	for num > 1 {
		if num%2 == 0 {
			factors = append(factors, 2)
			num = num / 2
		} else if num%3 == 0 {
			factors = append(factors, 3)
			num = num / 3
		} else {
			factors = append(factors, num)
			break
		}
	}
	return factors
}

func main() {
	factors := prime_factors(5)
	fmt.Println(factors)
}
